### PR TITLE
pre-PR for homing functionality + 402 state change callbacks

### PR DESCRIPTION
**Pull request by luminize**
This PR is to start feedback for bringing in Homing functionality and 402 state change callbacks
I've also added callbacks so that when a specific 402 profile state is reached, code can be executed. I use this mainly to push certain functions on a deque for async execution.

- I need to look into removing obsolete commented lines (I've spotted one at least)
- at this point I should still add some documentation additions

### CODE REVIEW

File: canopen/profiles/p402.py [hunk 1]
```diff
@@ -60,12 +89,26 @@ def setup_402_state_machine(self):
...
         self.pdo.tx[1].trans_type = 255
         self.pdo.tx[1].enabled = True
         self.pdo.tx[1].save()
         self.nmt.state = 'OPERATIONAL'
 
+    def add_callback_402_state(self, state, cb):
+        self.callbacks_402[state].append(cb)
+
+    def clear_402_callbacks(self):
+        print "clearing 402 callbacks"
```
Comment (christiansandberg) [inline canopen/profiles/p402.py]: print statement only works on Python 2. Either change to print(), remove it if not needed, or use logging module instead.

File: canopen/profiles/p402.py [hunk 2]
```diff
@@ -60,12 +89,26 @@ def setup_402_state_machine(self):
...
+        for key, (value, cblist) in enumerate(self.callbacks_402.iteritems()):
```
Comment (christiansandberg) [inline canopen/profiles/p402.py]: Oh, and ``.iteritems()`` should be changed to ``.items()`` to work on Python 3.

**Comment by baumann93** [comment]
Hello Christian, 
I want to use your module on windows10 with the CAN-Bus Adapter from VSCOM. Is there a possibility to use this adapter with your module on windows? 

I hope you can help me. 

Manuel

**Comment by christiansandberg** [comment]
It's OK to start a new issue if you have questions (I don't have a mailing list yet).

New interfaces are implemented in [python-can](https://github.com/hardbyte/python-can).

**Comment by christiansandberg** [comment]
There's been quite some work done on 402-functionality so I'm closing this for now.

**Pull request closed without merging**